Extensions docs update (#12491)
rancher/dashboard: docusaurus/docs/extensions/advanced/fix-annotations-published-extensions.md

# Fixing annotations on published extensions

> As the development of an extension progresses over time, through different versions of Rancher and the Shell package, looking back at those earlier releases some may think why is that extension version available for install in Rancher versions when you know it shouldn't work.

Reviewer: ```suggestion
As the development of an extension progresses over time, through different versions of Rancher and the Shell package, some users may wonder why a specific extension version is available for installation in Rancher versions when it is known that it shouldn't work.
```

Since that extension was already built and served to users, fixing annotations and repackaging an extension my be a pain point that would often get postponed.

With Shell `3.0.0` we have introduced a mechanism that will allow developers to fix annotations on those published extension version that will handle changes on `index.yaml` that will persist over time and also the re-packaging of the `tgz` assets with the correct annotations inside. To leverage this mechanism, developers should follow this procedure:

1) After identifying which extension version you want to fix the annotations for, go to branch of the repository where the Helm chart that serves your extension lives and edit the `Chart.yaml` in `charts/<-YOUR-EXT->/<-EXT-VERSION-TO-FIX->/Chart.yaml` and change the annotations there as desired

2) Once the annotations changes are done, on the root folder of the skeleton app that contains the extension just run the following command:
```js
yarn publish-pkgs -e -g

> ```

Reviewer: Just for clarification, does this mean pushing changes to the branch where the helm charts are served, then going back to the main branch to run the `publish-pkgs` script to see those changes?

Author: Your question raised me some concerns after reading this... There's no package.json on gh-pages branch... 🤔 I think I'll have to tweak this

Author: But to answer your question, you can't change branches otherwise it won't work... But I do get the point where your comment came from

Since that extension was already built and served to users, fixing annotations and repackaging an extension my be a pain point that would often get postponed.

> With Shell `3.0.0` we have introduced a mechanism that will allow developers to fix annotations on those published extension version that will handle changes on `index.yaml` that will persist over time and also the re-packaging of the `tgz` assets with the correct annotations inside (only works for Github). To leverage this mechanism, developers should follow this procedure:

Reviewer: ```suggestion
Starting with Shell version `3.0.0`, we have introduced a feature that allows developers to update annotations on published extension versions. This mechanism enables changes to persist in the `index.yaml` file over time and ensures that the `tgz` assets are re-packaged with the correct annotations (GitHub only). To leverage this mechanism, developers should follow this procedure:
```

> **1)** Copy over the contents of the root file `package.json` to have access to the yarn command alias `publish-pkgs`. You'll need to create the file on the branch where the Helm chart that serves your extension lives in

Reviewer: ```suggestion
**1)** Copy over the contents of the root file `package.json` to have access to the yarn command alias `publish-pkgs`. You'll need to create the file in the branch where the Helm chart that serves your extension lives in.
```

**2)** After identifying which extension version you want to fix the annotations for, go to branch of the repository where the Helm chart that serves your extension lives (for the examples we [provide](./../extensions-getting-started.md#creating-a-release) that branch should be `gh-pages`) and edit the `Chart.yaml` in `charts/<-YOUR-EXT->/<-EXT-VERSION-TO-FIX->/Chart.yaml` and change the annotations there as desired

> **3)** On the same branch, create on the root a file called `package.json`, with the contents copied from step 1)

Reviewer: ```suggestion
**3)** In the same branch, create a file named `package.json` in the root directory, using the contents copied from step 1.
```

**4)** Install node modules by running `yarn install`

> **5)** One the root of your branch, just run the following command:

Reviewer: ```suggestion
**5)** In the root of your branch, run the following command:
```

```js
yarn publish-pkgs -e -s <-GITHUB-ORG->/<-GITHUB-REPO->
```

In the case of the [Elemental repo](https://github.com/rancher/elemental-ui), the `GITHUB-ORG` would be `rancher` and `GITHUB-REPO` would be `elemental`

This will update the `index.yaml` of your Helm chart, propagating those annotation changes and also re-package the `tgz` file under `assets`.

> **6)** After the procedure is finished, just create a PR for your repository and merge your changes. The annotations that were changed in those published version should now be according to your needs. Careful to only commit the needed changes and not your new `package.json` or the `/node_modules` folder

Reviewer: ```suggestion
**6)** Once the procedure is complete, remove the `package.json`, `yarn.lock`, and `/node_modules` folder from your branch to avoid committing them.

**7)** Create a pull request for your repository and merge your changes. The annotations for the published version should now reflect your updates.
```
